### PR TITLE
Added a linear constant focusing element.

### PR DESCRIPTION
**Pull request by cemitch99**
This will be useful for future benchmarking with space charge.

### CODE REVIEW

File: src/particles/elements/ConstF.H [hunk 1]
```diff
@@ -0,0 +1,96 @@
+/* Copyright 2021 Chad Mitchell, Axel Huebl
```
Comment (ax3l) [inline src/particles/elements/ConstF.H]:
```suggestion
/* Copyright 2022 Chad Mitchell, Axel Huebl
```

File: src/ImpactX.cpp [hunk 2]
```diff
@@ -252,6 +252,13 @@ namespace impactx
                 pp_element.get("g", g);
                 pp_element.get("K2", K2);
                 m_lattice.emplace_back( DipEdge(psi, rc, g, K2) );
+            } else if (element_type == "constf") {
```
Comment (ax3l) [inline src/ImpactX.cpp:255]: That looks good.

Do you want to add a minimal input set that uses (aka: "covers") this code path already in continuous integration (CI)? Does not need to be super meaningful yet, but would make sure the code does not segfault as we continue to add code to the repo.

File: src/particles/elements/ConstF.H [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        }
+
+        /** This is a constf functor, so that a variable of this type can be used like a
+         *  constf function.
+         *
+         * @param p Particle AoS data for positions and cpu/id
+         * @param px particle momentum in x
+         * @param py particle momentum in y
+         * @param pt particle momentum in t
+         * @param refpart reference particle
```
Comment (ax3l) [inline src/particles/elements/ConstF.H]:
```suggestion
         * @param refpart reference particle (unused)
```

**Review by ax3l — COMMENTED** [review]
Looks great! :+1:

File: src/ImpactX.cpp [hunk 4]
```diff
@@ -252,6 +252,13 @@ namespace impactx
...
+                amrex::Real ds, kx, ky, kt;
+                pp_element.get("ds", ds);
```
Comment (ax3l) [inline src/ImpactX.cpp:257]: Please don't forget to add those to `docs/source/usage/parameters.rst`

File: src/ImpactX.cpp (hunk 2 above)
Comment (cemitch99) [reply src/ImpactX.cpp:255]: Yes, I can create another benchmark input file to test the consf element without space charge, which should work as-is.  Will work on this today.

File: src/ImpactX.cpp (hunk 4 above)
Comment (cemitch99) [reply src/ImpactX.cpp:257]: Will do this; thanks for the reminder.

File: src/ImpactX.cpp (hunk 2 above)
Comment (cemitch99) [reply src/ImpactX.cpp:255]: Just added a commit.  (See above.)

File: src/ImpactX.cpp (hunk 4 above)
Comment (cemitch99) [reply src/ImpactX.cpp:257]: Done.  (See above.)

File: examples/cfchannel/README.rst [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+The particle distribution should remain unchanged, to within the level expected due to numerical particle noise.
+This fact is independent of the length of the channel.  This is tested using the second moments of the distribution.
+
+In this test, the initial and final values of :math:`\sigma_x`, :math:`\sigma_y`, :math:`\sigma_t`, :math:`\epsilon_x`, :math:`\epsilon_y`, and :math:`\epsilon_t` must agree with nominal values.
+
+
+.. literalinclude:: input_cfchannel.in
+   :language: ini
+   :caption: You can copy this file from
+``examples/cfchannel/input_cfchannel.in``.
```
Comment (ax3l) [inline examples/cfchannel/README.rst]:
```suggestion
   :caption: You can copy this file from ``examples/cfchannel/input_cfchannel.in``.
```

File: examples/cfchannel/README.rst [hunk 6]
```diff
@@ -0,0 +1,28 @@
+.. _examples-cfchannel:
+
+Constant Focusing Channel
+==========================
```
Comment (ax3l) [inline examples/cfchannel/README.rst]: `.rst` is a bit picky with the alignment here:
```suggestion
Constant Focusing Channel
=========================
```

**Review by ax3l — APPROVED** [review]
Thank you 👍

**Pull request merged**